src/app/new-page/component/content-animation/content-animation.component.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  Component,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  SimpleChanges
+} from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { PageService } from '../../service/page-service.service';
+
+@Component({
+  selector: 'app-content-new-animation',
+  templateUrl: './content-animation.component.html',
+  styleUrls: ['./content-animation.component.css']
+})
+export class ContentAnimationNewComponent implements OnChanges, OnDestroy {
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input() item: any;
+
+  private _unsubscribeAll = new Subject<any>();
+  animation: any;
+  ready: boolean;
+  anmResource: string;
+  anmOptions: any;
+  anmViewItem: any;
+  hasEvents: boolean;
+  dir$: Observable<string>;
+
+  constructor(private pageService: PageService) {
+    this.dir$ = this.pageService.pageDir$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'item': {
+            if (
+              !changes['item'].previousValue ||
+              changes['item'].currentValue !== changes['item'].previousValue
+            ) {
+              this.ready = false;
+              this.animation = this.item;
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  onAnimationClick(): void {
+    if (this.animation.attributes.events) {
+      this.pageService.formAction(this.animation.attributes.events);
+    }
+  }
+
+  onAnimationCreated(anim: any) {
+    this.anmViewItem = anim;
+  }
+
+  private init(): void {
+    this.anmResource = this.pageService.getAnimationUrl(
+      this.animation.attributes.resource
+        ? this.animation.attributes.resource.trim()
+        : ''
+    );
+    if (this.anmResource) {
+      this.anmOptions = {
+        path: this.anmResource,
+        loop: !!this.animation.attributes.loop,
+        autoplay: !!this.animation.attributes.autoplay
+      };
+    }
+
+    this.hasEvents = !!this.animation.attributes.events;
+
+    if (
+      !!this.animation.attributes.playListeners ||
+      !!this.animation.attributes.stopListeners
+    ) {
+      this.awaitAnimEvent();
+    }
+
+    this.ready = true;
+  }
+
+  private awaitAnimEvent(): void {
+    this.pageService.contentEvent$
+      .pipe(takeUntil(this._unsubscribeAll))
+      .subscribe((event) => {
+        if (this.animation.attributes.playListeners) {
+          const tEvents = this.animation.attributes.playListeners.split(' ');
```

Comment:
are you having to convert the array of `EventIds` into a space separate list to just re-split them here?

Reply:
I reworked this page. Somehow I missed testing/building this component with the new parser. I've rebuilt this entire page to work with the Parser now.